llarp/handlers/tun.cpp

```diff
@@ -857,9 +930,7 @@ namespace llarp
       TunEndpoint *self = static_cast< TunEndpoint * >(tun->user);
       ManagedBuffer buf(b);
       if(!self->m_UserToNetworkPktQueue.EmplaceIf(
-             [buf](net::IPv4Packet &pkt) -> bool {
-               return pkt.Load(buf) && pkt.Header()->version == 4;
-             }))
+             [buf](net::IPPacket &pkt) -> bool { return pkt.Load(buf); }))
```

Comment:
If I'm reading the docs right (I'm a bit rusty on lambdas) then this is capturing `buf` by-copy as opposed to by-reference.  Is this desired?

Reply:
you're right it should be by reference

Reply:
ammended